### PR TITLE
prow: trigger: skip jobs whose parents are skipped

### PR DESCRIPTION
**Pull request by stevekuznetsov**
We must provide a status for every ProwJob that is configured and not
marked to be silent, both so that developers can see what happened to
their PR as well as so that tide can determine what the status of the PR
is. Therefore, we have to mark contexts that are children of skipped
contexts as also skipped.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/area prow
/kind bug
/cc @BenTheElder @krzyzacy @munnerz 
/assign @cjwagner 
fixes https://github.com/kubernetes/test-infra/issues/8665

### CODE REVIEW

**Comment by munnerz** [comment]
This looks like it'll fix our issues (as I discussed with you quite a while ago @BenTheElder!).

I've not had a chance to properly roll this out to our own infra and test the patch, but it looks fairly straight forward to me.

Adding my lgtm, but holding to allow someone more familiar to approve/merge!

/lgtm
/hold

File: prow/plugins/trigger/trigger.go [hunk 1]
```diff
@@ -241,6 +241,12 @@ func runOrSkipRequested(c client, pr *github.PullRequest, requestedJobs []config
 			toRun.Insert(job.Context)
 		} else if !job.SkipReport {
 			toSkip.Insert(job.Context)
+			// we need to post context statuses for all jobs; if a job is slated to
+			// run after the success of a parent job that is skipped, it must be
+			// skipped as well
+			for _, child := range job.RunAfterSuccess {
```
Comment (BenTheElder) [inline prow/plugins/trigger/trigger.go]: This doesn't appear to handle children of children?

File: prow/plugins/trigger/trigger.go [hunk 2]
```diff
@@ -241,6 +249,12 @@ func runOrSkipRequested(c client, pr *github.PullRequest, requestedJobs []config
...
```
Comment (munnerz) [inline prow/plugins/trigger/trigger.go]: Can this whole block now be replaced with a single `toSkip.Insert(allContexts(job.Context))`?

**Comment by stevekuznetsov** [comment]
@munnerz @BenTheElder updated -- infinitely recursive `run_after_success` is so yucky...

**Comment by munnerz** [comment]
:this_is_fine: 🙄

/lgtm

**Comment by stevekuznetsov** [comment]
@BenTheElder is this known

```
E0712 20:07:03.734] unexpected error
Traceback (most recent call last):
  File "./test-infra/jenkins/bootstrap.py", line 1095, in bootstrap
  File "./test-infra/jenkins/bootstrap.py", line 887, in job_script
ValueError: ('cannot find scenario for job', 'pull-test-infra-verify-bazel')
```

/retest

**Comment by krzyzacy** [comment]
hummmmm that means it used an old config for whatever reason

**Pull request merged**